packages/babel-parser/src/parser/expression.ts

```diff
@@ -2397,6 +2426,7 @@ export default class ExpressionParser extends LValParser {
   parsePropertyName(
     this: Parser,
     prop: Undone<
+      // @ts-expect-error todo(flow->ts) Property 'type' is missing in type 'TsNamedTypeElementBase' but required in type 'Node'.
       N.ObjectOrClassMember | N.ClassMember | N.TsNamedTypeElementBase
```

Comment:
Does `Undone<N.ObjectOrClassMember | N.ClassMember> | N.TsNamedTypeElementBase` work?

Reply:
the issue here is that `Undone` has a constraint for type argument `T extends NodeType`, which requires `type` field to be present… but that is missing in `TsNamedTypeElementBase`

Reply:
Yeah, I mean `NodePath<...> | N.TsNamedTypeElementBase`. `N.TsNamedTypeElementBase` does not have `type`, either. Or we can search which types are extending `TsNamedTypeElementBase` and use `Undone<those types>` instead.

Reply:
oh, misread the comment - indeed that works.
updated the pr